proposals/public-suffix.md

```diff
@@ -0,0 +1,227 @@
+*registrable domain / eTLD+1*s from domain names / URLs.
+
+Secondary objectives of this API are to:
+
+1. Improve extension developer experience by reducing complexity and maintenance overhead,
+since developers will no longer need to roll their own solutions for obtaining and parsing
+the [Public Suffix List (PSL)](https://publicsuffix.org/list/).
+2. Reduce extensions' resource usage (CPU, memory, disk space), since extensions
+will no longer be duplicating work already done by the host browser.
+
+#### Use Case #1: Wildcard Domains
+
+This API is relevant to any extension that gives the user control over which
+domain names / URLs the extension's functionality should apply to. For example,
+Mozilla's [multi-account-containers](https://github.com/mozilla/multi-account-containers)
+extension allows users to create containers for web pages as they visit them.
+
+Users may want the ability to apply the extension's functionality to a
+*wildcard domain*, such that all subdomains of some base domain are
+automatically included.
+
+This API allows the extension to automatically propose the topmost *registrable domain / eTLD+1*
+as a possible *wildcard domain* for the user to choose. It also allows the
+extension to prevent the user from accidentally choosing a *public suffix / eTLD*
+as a wildcard domain, e.g. `*.com`.
+
+#### Use Case #2: Grouping Domains in UI
+
+Where extensions present lists of domain names / URLs to users, it can be beneficial
+from a UX perspective to group them by their topmost *registrable domain / eTLD+1*s.
+
+##### Ungrouped domains
+
+|                       |
+| --------------------- |
+| example.co.uk         |
+| example2.com          |
+| foo.bar.example.co.uk |
+| foo.bar.example2.com  |
+| www.example.co.uk     |
+| www.example2.com      |
+
+##### Grouped domains
+
+| example.co.uk  |                       |
+| -------------- | --------------------- |
+|                | example.co.uk         |
+|                | foo.bar.example.co.uk |
+|                | www.example.co.uk     |
+
+| example2.com   |                       |
+| -------------- | --------------------- |
+|                | example2.com          |
+|                | foo.bar.example2.com  |
+|                | www.example2.com      |
+
+### Known Consumers
+
+Mozilla intends to make use of this API in its [multi-account-containers](https://github.com/mozilla/multi-account-containers) extension to allow users to create containers
+that include wildcard domain names: [PR #2352](https://github.com/mozilla/multi-account-containers/pull/2352). Currently, users must manually curate their containers to capture all
+anticipated subdomains of a base domain, and this manual approach is cumbersome and
+error-prone.
+
+Apart from this, any other extension that already rolls its own solution in order to parse
+the PSL could potentially benefit from this API, for example [uBlock Origin](https://github.com/gorhill/uBlock).
```

Comment:
Could you be more specific in the specific use cases that depend on this proposed API? Knowing the use cases makes it easier to evaluate the proposed API.

Reply:
Agreed, the more use cases the better. I will take a look at the source code of other extensions to see how they are using the PSL data currently.

Reply:
Done 3301526fcb150706aa59e9f20839d646e4920649